OSSM-8770 Update feature support tables (#88965)
openshift/openshift-docs: modules/ossm-release-notes-istio-deployment-lifecycle.adoc

////
Module included in the following assemblies:
* service-mesh-docs-main/ossm-release-notes-support-tables-assembly.adoc
////

//Post GA: might be worth considering putting tables into their own modules. Might easier to maintain. Warrants further discussion post GA.

:_mod-docs-content-type: REFERENCE
[id="istio-deployment-and-lifecycle_{context}"]
= Istio deployment and lifecycle

[cols="1,1"]
|===
| Feature | Status

| Installation with the {SMProductName} Operator
| GA

| Istio sidecar mode data plane
| GA

> | `InPlace` and `RevisionBased` control plane upgrades with the {SMProductName} Operator

Reviewer: @longmuir  Does it make sense to also include `IstioRevisionTag` support and istioctl tool?

Reviewer 2: `IstioRevisionTag` is listed as a supported API in the Sail Operator section, so this entry is more about the two types of upgrade paths. 

It's a good point about Istioctl though, I missed that we didn't have an entry for it.

@gwynnemonahan Could we add an entry at the bottom (or second from the bottom above the "Helm or Istioctl Installation" entry), that lists "Istioctl for select commands" as a GA feature?

Author: > @gwynnemonahan Could we add an entry at the bottom (or second from the bottom above the "Helm or Istioctl Installation" entry), that lists "Istioctl for select commands" as a GA feature?

@longmuir will Istioctl always just be for select commands? or will there be a time where it is just "Istioctl support"? Asking as users may wonder what "select commands" are supported, and need to be direct accordingly.

Reviewer 2: The istio-csr doc page defines which commands are supported with a separate table. It's unlikely we'll ever support all Istioctl commands. I guess it could also be worded "Istioctl for a selection of commands".

This link seems to work: https://88965--ocpdocs-pr.netlify.app/openshift-service-mesh/latest/install/ossm-istioctl-tool Not sure if it's possible to link to that page.

Reviewer 2: FWIW, I notice in the new features notes we say Istiocti is supported "for select platforms and commands" Which also works.

////
Module included in the following assemblies:
* service-mesh-docs-main/ossm-release-notes-support-tables-assembly.adoc
////

> //Post GA: might be worth considering putting tables into their own modules. Might easier to maintain. Warrants further discussion post GA.

Reviewer 3: Just wondering if this note can be removed throughout the modules, since it seems you did move them into their own modules?

:_mod-docs-content-type: REFERENCE
[id="istio-deployment-and-lifecycle_{context}"]
= Istio deployment and lifecycle

[cols="1,1"]
|===
| Feature | Status

| Installation with the {SMProductName} Operator
| GA

| Istio sidecar mode data plane
| GA

| `InPlace` and `RevisionBased` control plane upgrades with the {SMProductName} Operator
| GA
| The Istio multicluster mesh deployment models
| GA

| The Istio external control plane deployment models
| GA

> | Multiple control planes on a single OpenShift cluster

Reviewer 3: "OpenShift" shouldn't be alone normally - I assume should be: 
```suggestion
| Multiple control planes on a single {ocp-product-title} cluster
```

| GA

> | `IstioCNI` Plugin | GA

Reviewer 3: ```suggestion
| `IstioCNI` plugin | GA
```

| Istio configuration scoping: Sidecar API, `exportTo` and Discovery Selectors
| GA

> | IPv6 Support

Reviewer 3: ```suggestion
| IPv6 support
```

| GA

> | Dual Stack IPv4/IPv6

Reviewer 3: ```suggestion
| Dual stack IPv4/IPv6
```

| TP

> | Virtual Machine (non-OpenShift) workload integration

Reviewer 3: Again shouldn't have just "OpenShift" here. Not sure what it should be 

```suggestion
| Virtual machine (non-OpenShift) workload integration
```

| NA

| Istioctl for select commands
| GA ^[1]^

> | Helm or Istioctl Installation

Reviewer 3: ```suggestion
| Helm or Istioctl installation
```

| NA ^[2]^

| ProxyConfig
| GA ^[3]^
|===

> . For more information, see: "Support for Istioctl".

Reviewer 3: ```suggestion
. For more information, see "Support for Istioctl".
```

> . Installation is only supported via the {SMProduct} 3 Operator, which uses the Istio Helm chart values for managing configuration.

Reviewer 3: Shouldn't use "via" per ISG

```suggestion
. Installation is only supported by using the {SMProduct} 3 Operator, which uses the Istio Helm chart values for managing configuration.
```
